lib/iris/coords.py

```diff
@@ -1762,30 +1761,42 @@ def data(self, data):
         if data is None:
             raise ValueError('The data payload of a CellMeasure may not be '
                              'None; it must be a numpy array or equivalent.')
+        if is_lazy_data(data) and data.dtype.kind in 'iu':
+            # Disallow lazy integer data.  Integer values would be very
+            # unexpected anyway, but this avoids the *even more obscure*
+            # possibility that there are masked values, because masked integer
+            # data would cause dask to return data with the wrong dtype.
+            # This can all be fixed, as for cube data, by adding fill_value and
+            # dtype support, but for now we prefer just not to support this.
+            msg = ('Cannot create cell measure with lazy data of type {}, as '
+                   'integer types are not currently supported.')
+            raise ValueError(msg.format(data.dtype))
         if data.shape == ():
+            # If we have a scalar value, promote the shape from () to (1,).
+            # NOTE: this way also *realises* it.  Don't think that matters.
             data = np.array(data, ndmin=1)
-        if hasattr(self, '_data') and self._data is not None:
+        if hasattr(self, '_data_manager') and self._data_manager is not None:
             # Check that setting these data wouldn't change self.shape
             if data.shape != self.shape:
                 raise ValueError("New data shape must match existing data "
                                  "shape.")
 
-        self._data = data
+        self._data_manager = DataManager(data)
```

Comment:
I'll add comments in all of `__init__`, `copy` and `__getitem__` to make clear how it all fits together.